bin/oneoff/backfill_data/school_info_id

```diff
@@ -15,39 +15,38 @@ SchoolInfo.find_each(batch_size: 2500) do |school_info|
 end
 puts "\nPROCESSED SCHOOL_INFO"
 
-batch_number = 0
-User.with_deleted.find_in_batches(batch_size: 2_500) do |batch|
-  puts "PROCESSING USERS: #{batch_number}..."
-  ActiveRecord::Base.transaction do
-    batch.each do |user|
-      # only do this if user has no school_info_id or their school_info_id is
-      # not associated with a school_id
-      # (don't do it if the user already has a school_info_id associated with a school_id)
-      next if school_info_id_school_id_mapping[user.school_info_id]
-
-      enrollments = Pd::Enrollment.with_deleted.for_user(user)
-      next if enrollments.empty?
-
-      enrollments_with_school_id = enrollments.select {|e| school_info_id_school_id_mapping[e.school_info_id]}
-
-      # if the user has pd_enrollments whose school_info_id is associated with
-      # a school_id, update the user with the most recent of those pd_enrollments' school_info_id
-      if enrollments_with_school_id.any?
-        enrollment = enrollments_with_school_id.sort_by {|e| e[:updated_at]}.last
-        user.school_info_id = enrollment.school_info_id
-        user.save!(validate: false) # skip validations because many users don't pass validation
-      # if user doesn't have a school_info_id at all, update with most recent
-      # enrollment school_info_id even though it doesn't have school_id
-      elsif user.school_info_id.nil?
-        enrollment = enrollments.sort_by {|e| e[:updated_at]}.last
-        user.school_info_id = enrollment.school_info_id
-        user.save!(validate: false) # skip validations because many users don't pass validation
-      end
-      # if the user has a school_info_id not associated with a school_id, and
-      # none of their pd_enrollments' school_info_ids are associated with a
-      # school_id, don't do anything
-    end
+puts "\nGETTING USER_IDS"
+user_ids = Pd::Enrollment.with_deleted.select(:user_id).distinct.pluck(:user_id)
+
+puts "\nPROCESSING USERS"
+user_ids.each do |user_id|
+  user = User.with_deleted.find(user_id)
+  # only do this if user has no school_info_id or their school_info_id is
+  # not associated with a school_id
+  # (don't do it if the user already has a school_info_id associated with a school_id)
+  next if school_info_id_school_id_mapping[user.school_info_id]
+
+  enrollments = Pd::Enrollment.with_deleted.for_user(user)
+  next if enrollments.empty?
+
+  enrollments_with_school_id = enrollments.select {|e| school_info_id_school_id_mapping[e.school_info_id]}
+
+  # if the user has pd_enrollments whose school_info_id is associated with
+  # a school_id, update the user with the most recent of those pd_enrollments' school_info_id
+  if enrollments_with_school_id.any?
+    enrollment = enrollments_with_school_id.sort_by {|e| e[:updated_at]}.last
+    user.school_info_id = enrollment.school_info_id
+    user.save!(validate: false) # skip validations because many users don't pass validation
```

Comment:
If all we're doing is updating the `school_info_id` column, it might be better to use [update_column](https://apidock.com/rails/v3.1.0/ActiveRecord/Persistence/update_column) instead of these 2 lines.

e.g. 
```ruby
user.update_column(:school_info_id, enrollment.school_info_id)
```
This will update the column directly in the DB, skipping validation.